File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/AutocompleteTextAtCaret.java
package com.sourcegraph.cody.autocomplete;

import com.intellij.openapi.editor.Caret;
import org.jetbrains.annotations.NotNull;

public class AutocompleteTextAtCaret {
  @NotNull public final AutocompleteText autoCompleteText;
  @NotNull public final Caret caret;

  public AutocompleteTextAtCaret(@NotNull Caret caret, @NotNull AutocompleteText autoCompleteText) {
    this.caret = caret;
    this.autoCompleteText = autoCompleteText;
  }

  public AutocompleteTextAtCaret(
      @NotNull Caret caret,
      @NotNull String sameLineBeforeSuffixText,
      @NotNull String sameLineAfterSuffixText,
      @NotNull String blockText) {
    this(caret, new AutocompleteText(sameLineBeforeSuffixText, sameLineAfterSuffixText, blockText));
  }
}

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/autocomplete/AutocompleteTextAtCaret.kt
package com.sourcegraph.cody.autocomplete

import com.intellij.openapi.editor.Caret

class AutocompleteTextAtCaret(val caret: Caret, val autoCompleteText: AutocompleteText) {
  constructor(
      caret: Caret,
      sameLineBeforeSuffixText: String,
      sameLineAfterSuffixText: String,
      blockText: String
  ) : this(caret, AutocompleteText(sameLineBeforeSuffixText, sameLineAfterSuffixText, blockText))
}
